js/viewmodel.js

```diff
@@ -7,7 +7,30 @@ window.FBE = window.FBE || {};
     var events = new TinyEmitter(),
         bp = new Blueprint(),
         selectedItem,
-        entityNamesThatDoNotRotate = /beacon|roboport|lab|heat_pipe|pipe$|furnace|chest|pole|substation|solar|accumulator|centrifuge|wall|rocket_silo|radar|turret|speaker|power_switch|reactor|lamp|land_mine/i
+        entityNamesThatDoNotRotate = /beacon|roboport|lab|heat_pipe|pipe$|furnace|chest|pole|substation|solar|accumulator|centrifuge|wall|rocket_silo|radar|turret|speaker|power_switch|reactor|lamp|land_mine/i,

```

Comment:
These regexes still feel wrong, as they are just a list of options. i think these would be better handled by an array.

Also doing a little bit of searching, i think this regex match performs a lot slower than using indexOf

Reply:
I believe Array.includes() is what you should be using to check the existence of a string in an array

Reply:
This regex is called once at app startup and whenever we load a blueprint. I don't think it's a performance bottleneck.

Profiling on my aging laptop (i5, chrome, ubunut) it takes 5ms to call `getPlaceableItems`, and of that, 0.16ms is spent in `entityNamesThatDoNotRotate.test`.

Is that fast enough?

Reply:
right, but this still feels like an abuse of regex. It's being used like an array, instead of using an array.

Reply:
Will merge this in for now, but opened #54 to get this fixed